contrib/cirrus/runner.sh

```diff
@@ -407,8 +408,21 @@ dotest() {
         die "Found fallback podman '$fallback_podman' in \$PATH; tests require none, as a guarantee that we're testing the right binary."
     fi
 
+    # Catch invalid "TMPDIR == /tmp" assumptions; PR #19281
+    TMPDIR=$(mktemp --tmpdir -d CI_XXXX)
+    # tmp dir is commonly 1777 to allow all user to read/write
+    chmod 1777 $TMPDIR
+    export TMPDIR
+    fstype=$(findmnt -n -o FSTYPE --target $TMPDIR)
+    if [[ "$fstype" != "tmpfs" ]]; then
+        die "The CI test TMPDIR is not on a tmpfs mount, we need tmpfs to make the tests faster"
```

Comment:
Not worth a re-push, but: in case of failure, this will be very difficult to debug. A more helpful message might be something like
```bash
    die "The CI test TMPDIR ($TMPDIR) fs type is '$fstype'; it should be 'tmpfs' (PR #22533)"
```